### PR TITLE
Add missing tests for edge cases

### PR DESCRIPTION
**Pull request by kabirbaidhya**
Add missing tests for edge cases for better coverage.

### CODE REVIEW

File: test/domain.test.ts [hunk 1]
```diff
@@ -13,6 +14,44 @@ describe('store: [adapter=DOMAIN]', () => {
     Object.assign(process, { domain: undefined });
   });
 
+  describe('initialize()', () => {
+    it('should initialize the store.', done => {
+      const callback = () => {
+        expect(!!(process.domain as any)[STORE_KEY]).to.equal(true);
+        expect(globalStore.isInitialized()).to.equal(true);
+        done();
```
Comment (mesaugat) [inline test/domain.test.ts]: New line before done.

File: test/domain.test.ts [hunk 2]
```diff
@@ -13,6 +14,44 @@ describe('store: [adapter=DOMAIN]', () => {
...
+
+    it('should also bind params if params are passed through arguments.', done => {
+      const req = new EventEmitter();
+      const res = new EventEmitter();
+      const emitters = [req, res];
+      const errorCallback = (err: any) => err;
+
+      const callback = () => {
+        expect(!!(process.domain as any)[STORE_KEY]).to.equal(true);
+        expect(globalStore.isInitialized()).to.equal(true);
```
Comment (mesaugat) [inline test/domain.test.ts]: This part of the assertion is already in `it('should initialize the store.', ...`. I don't think we need to do the same here. If the first test passes, this should pass as well.

File: test/domain.test.ts [hunk 3]
```diff
@@ -44,6 +83,16 @@ describe('store: [adapter=DOMAIN]', () => {
       expect(globalStore.get.bind(globalStore, 'foo')).to.throw('No active domain found in store.');
     });
 
+    it('should return null if invoked under active domain w/o proper store initialization.', done => {
+      const d = domain.create();
+
+      d.run(() => {
+        // Ensure data in the existing domain is available at this point.
+        expect(globalStore.get('foo')).to.equal(null);
+        done();
```
Comment (mesaugat) [inline test/domain.test.ts]: .

File: test/domain.test.ts [hunk 4]
```diff
@@ -55,7 +104,7 @@ describe('store: [adapter=DOMAIN]', () => {
   });
 
   describe('getAll()', () => {
-    it('should return all value from the store', done => {
+    it('should return all value from the store.', done => {
```
Comment (mesaugat) [inline test/domain.test.ts]:
```diff
- value
+ values
```

File: test/domain.test.ts [hunk 5]
```diff
@@ -80,6 +129,16 @@ describe('store: [adapter=DOMAIN]', () => {
...
+
+    it('should return null if invoked under active domain w/o proper store initialization.', done => {
+      const d = domain.create();
+
+      d.run(() => {
+        // Ensure data in the existing domain is available at this point.
+        expect(globalStore.getAll()).to.equal(null);
+        done();
+      });
+    });
```
Comment (mesaugat) [inline test/domain.test.ts]: Duplicate test case. This is the same as the one above.

File: test/domain.test.ts [hunk 6]
```diff
@@ -238,7 +297,21 @@ describe('store: [adapter=DOMAIN]', () => {
...
-    it('should set properties in the store', done => {
+    it('should throw an error if invalid arguments are provided.', done => {
+      const callback = () => {
+        expect(globalStore.set.bind(globalStore, 5)).to.throw('Invalid arguments provided for asyncStore.set()');
+        expect(globalStore.set.bind(globalStore, 'five')).to.throw('Invalid arguments provided for asyncStore.set()');
+        expect(globalStore.set.bind(globalStore, null)).to.throw('Invalid arguments provided for asyncStore.set()');
+        expect(globalStore.set.bind(globalStore, undefined)).to.throw(
+          'Invalid arguments provided for asyncStore.set()'
+        );
+        done();
```
Comment (mesaugat) [inline test/domain.test.ts]: .

File: test/general.test.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import 'mocha';
+import { expect } from 'chai';
+import * as globalStore from '../src';
```
Comment (mesaugat) [inline test/general.test.ts]: .

File: test/domain.test.ts (hunk 5 above)
Comment (kabirbaidhya) [reply test/domain.test.ts]: The previous one was done for `get()` this one is for `getAll()` they are identical tests but for different functions.

Comment (mesaugat) [reply test/domain.test.ts]: Didn't realize it.
